pkg/workceptor/workunitbase.go

```diff
@@ -318,6 +320,8 @@ func (bwu *BaseWorkUnit) monitorLocalStatus() {
 		select {
 		case event := <-watcherEvents:
 			if event.Op&fsnotify.Write == fsnotify.Write {
+				bwu.statusLock.RLock()
```

Comment:
Seems like this should be a full lock, not just a read lock, since we're writing to the in-memory data structure.

Reply:
Oh? I assumed we only wanted a full lock when writing to the file. I must have misunderstood how these were being used.

Reply:
The lock object protects the in-memory data structure.  When we're loading from the file, we're writing to the copy in memory.